x-pack/plugins/fleet/common/services/validate_package_policy.ts

```diff
@@ -357,6 +367,16 @@ export const validatePackagePolicyConfig = (
     }
   }
 
+  if (varName === DATASET_VAR_NAME && packageType === 'input' && parsedValue !== undefined) {
+    const { valid, error } = isValidDataset(
+      parsedValue.dataset ? parsedValue.dataset : parsedValue,
```

Comment:
added a change to let the validation work on API calls with invalid dataset, it seems in API call, the dataset value is a simple string, not an object

tested by calling the API from kibana console:
```
POST kbn:/api/fleet/package_policies
{
    "name": "log-7",
    "description": "",
    "namespace": "",
    "policy_id": "b37b2a53-96a2-460c-8a7e-891ee69f2692",
    "enabled": true,
    "inputs": [
        {
            "type": "logfile",
            "policy_template": "logs",
            "enabled": true,
            "streams": [
                {
                    "enabled": true,
                    "data_stream": {
                        "type": "logs",
                        "dataset": "log.logs"
                    },
                    "vars": {
                        "paths": {
                            "type": "text",
                            "value": [
                                "/"
                            ]
                        },
                        "data_stream.dataset": {
                            "type": "text",
                            "value": "generic*"
                        }
                    }
                }
            ]
        }
    ],
    "package": {
        "name": "log",
        "title": "Custom Logs",
        "version": "2.3.1"
    },
    "force": false
}

{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "Package policy is invalid: inputs.logfile.streams.log.logs.vars.data_stream.dataset: Dataset contains invalid characters"
}
```

Reply:
Do you think that this test could be added to the functional tests for this endpoint?